### ISSUE TITLE
False Positive: VariableDeclarationUsageDistance Check

### ISSUE THREAD

**Issue by Vyom-Yadav** [checkstyle/checkstyle#13154]
I have read check documentation: https://checkstyle.sourceforge.io/config_coding.html#VariableDeclarationUsageDistance
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
/var/tmp $ javac Test.java

/var/tmp $ cat config.xml
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.2//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_2.dtd">

<module name="Checker">
    <module name="TreeWalker">
        <module name="VariableDeclarationUsageDistance">
            <property name="allowedDistance" value="1"/>
            <property name="validateBetweenScopes" value="true"/>
        </module>
    </module>
</module>

/var/tmp $ cat Test.java
public class Test {
    private void method5() {
        Object[] values = {}; // no violation expected
        for (int i = 0; i < 10; i++) {
            if (true) {
                System.out.println();
                Object b = values[i];
            }
        }
    }
}


/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-10.12.1-SNAPSHOT-all.jar -c config.xml Test.java
Starting audit...
[ERROR] Test.java:3:9: Distance between variable 'values' declaration and its first usage is 2, but allowed 1.  Consider making that variable final if you still need to store its value in advance (before method calls that might have side effects on the original value). [VariableDeclarationUsageDistance]
Audit done.
Checkstyle ends with 1 errors.

```

---

**Describe what you expect in detail.**

There is no possible way to reduce the distance between declaration and usage. The check wrongly calculates the distance as 2.

**Comment by Vyom-Yadav** [comment]
This is the behavior after the merger of https://github.com/checkstyle/checkstyle/pull/11672
See https://github.com/checkstyle/checkstyle/pull/11672#discussion_r1195149755

**Comment by romani** [comment]
@Vyom-Yadav , there is big problem with this ....
we try to fix false negative (absence of violation) and cause false positive (violation).
From user perspective: there was no checkstyle violation , CI is green, yes there are some not found not-ideal code but it is ok. User update to new version and he has violations that are not possible to resolve, so user need to revert checkstyle version bump or disable a Check.

Do you really think it is good ?

---------

We can agree on false positive only if existed before, and we just noticed it in recent update but it existed even before.

**Comment by Z4ID-Reaper** [comment]
@romani @Zopsss 
I have read check documentation: https://checkstyle.sourceforge.io/config_coding.html#VariableDeclarationUsageDistance
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

though the issue seems to be solved.
i don't know why there are so many variations in errors in case of "VariableDeclarationUsageCheck"
we need to take a look at all of them at once and then only we should remove issue that seems to be solved.

note : all the files used were same as the reporter of the issue above

### config.xml

```
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.2//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_2.dtd">

<module name="Checker">
    <module name="TreeWalker">
        <module name="VariableDeclarationUsageDistance">
            <property name="allowedDistance" value="1"/>
            <property name="validateBetweenScopes" value="true"/>
        </module>
    </module>
</module>
```

---
### Test.java

```
public class Test {
    private void method5() {
        Object[] values = {}; // no violation expected
        for (int i = 0; i < 10; i++) {
            if (true) {
                System.out.println();
                Object b = values[i];
            }
        }
    }
}
```
---
### cli output:

```
D:\zaid\zaid\codes\asasas>java -jar checkstyle-10.23.0-all.jar -c config.xml Test.java
Starting audit...
Audit done.
```

⚠  no error was expected hence the issue seems to be solved

**Comment by romani** [comment]
confirmed, no violations on latest, issue might be resovled in scope of some other update.
